packages/components/select/src/js/select.controller.js

```diff
@@ -78,12 +78,21 @@ export default class {
     }
   }
 
-  removeChoice(e, index, callback) {
+  removeChoice(e, index, item, callback) {
     e.preventDefault();
     e.stopPropagation();
 
-    // Call $selectMultiple.removeChoice of ui-select
-    callback(index, this);
+    if (this.onConfirmRemove) {
```

Comment:
But when I use it in Manager, I can pass a string value into the component which could potentially break the component right?

Reply:
Whatever property is passed it will still be a function on the directive side because it is declared as an optional expression. We just have to know if it is defined

Template
```javascript
on-confirm-remove="return 42"
```
Directive
```javascript
const value = onConfirmRemove && onConfirmRemove() // 42
```

Reply:
> The & binding allows a directive to trigger evaluation of an expression in the context of the original scope, at a specific time. Any legal expression is allowed, including an expression which contains a function call. Because of this, & bindings are ideal for binding callback functions to directive behaviors.

https://docs.angularjs.org/guide/directive